src/Raven.Studio/typescript/viewmodels/manage/infoPackage.ts

```diff
@@ -47,23 +60,58 @@ class infoPackage extends viewModelBase {
     }
 
     private startDownload(url: string) {
-        const $form = $("#downloadInfoPackageForm");
-        
-        
         this.getNextOperationId()
             .done((operationId: number) => {
+                this.operationId = operationId;
                 const operationPart = "?operationId=" + operationId;
-                $form.attr("action", appUrl.baseUrl + url);
+
+                const $form = $("#downloadInfoPackageForm");
+                $form.attr("action", appUrl.baseUrl + url + operationPart);
                 $("[name=operationId]", $form).val(operationId.toString());
                 $form.submit();
 
                 notificationCenter.instance.monitorOperation(null, operationId)
                     .always(() => {
                         this.spinners.clusterWide(false);
                         this.spinners.serverWide(false);
+                        this.spinners.abort(false);
                     });
             });
     }
+
+    private confirmLeavingPage() {
+        const abort = "Leave and Abort";
+        const stay = "Stay on this page";
+        const abortResult = $.Deferred<confirmDialogResult>();
+
+        const confirmation = this.confirmationMessage("Abort Debug Package Creation", "Leaving this page will abort package creation.<br>How do you want to proceed?", {
```

Comment:
where do we have code which aborts package creation?

Reply:
The cancelation token contains the `HttpContext.RequestAborted` param
Once leaving the page, `HttpContext.RequestAborted` becomes true